### PR TITLE
Tests

### LINKED ISSUE #1: Tests!
Most of the code was extracted from chapman, but chapman didn't have tests for either AMQP::Ping or AMQP::Connection :(

### PR DESCRIPTION
**Pull request by roback**
close #1 

Used [the tests at bunny](https://github.com/ruby-amqp/bunny/blob/master/spec/higher_level_api/integration/basic_publish_spec.rb) as inspiration.

### TODO

- [x] Connection tests
- [x] Ping tests
- [x] Subscription tests
- [x] Rabbitmq on travis (d6af208fe12f51846ff3a04ff89d8674a4fd10fc)
- [x] TLS (#15) (https://github.com/twingly/twingly-amqp/commit/4c98e0cb4081ebc670707070f300183dfe1ccf4d)

The `Subscription#subscribe` locks the thread so I'm not sure on how to do the testing, but I'll try to come up with something tomorrow :)

### CODE REVIEW

**Comment by roback** [comment]
> TLS

Created a test for this and got the following error from bunny:

```
Bunny::TCPConnectionFailedForAllHosts:
       Could not establish TCP connection to any of the configured hosts
```

The only way I can think of testing `Twingly::AMQP::Connection.new` is to check whether it throws the above error.

**Comment by jage** [comment]
> The only way I can think of testing Twingly::AMQP::Connection.new is to check whether it throws the above error.

I guess it's better than no test?

**Comment by roback** [comment]
> I guess it's better than no test?

I suppose it is :)

File: spec/twingly/ping_spec.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+require "amqp_queue_context"
+
+describe Twingly::AMQP::Ping do
+  include_context "amqp queue"
```
Comment (jage) [inline spec/twingly/ping_spec.rb]: Nice use of context!

Comment (roback) [inline spec/twingly/ping_spec.rb]: I just had to try it :)

File: spec/twingly/connection_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+describe Twingly::AMQP::Connection do
+  describe ".new" do
+    let(:host) { "localhost" }
+    let(:rabbitmq_host_env_name) { "RABBITMQ_01_HOST" }
+
+    context "without arguments" do
+      it "should read them from ENV" do
+        # TODO test doesnt really check that they are read from ENV
```
Comment (jage) [inline spec/twingly/connection_spec.rb]: Should we try and fix this TODO, or maybe create an issue instead?

Comment (roback) [inline spec/twingly/connection_spec.rb]: Created an issue because I don't know how to test this correctly. #16

**Comment by roback** [comment]
@jage Whats the status on this one? :)

**Comment by jage** [comment]
I think we should merge, I find it hard to improve the specs currently. It will be much easier when we use this gem in more projects.

**Comment by roback** [comment]
I'll merge then, but I wait with bumping the version until I have solved some more issues.

**Comment by jage** [comment]
:+1:

**Pull request merged**